Add columns - FunctionInvocationId, ScriptHostInstanceId and HostId

pragnagopa opened the pull request:
Fixes #2312 
Updated all the Tracewriter.Info and Tracewriter.Error calls to include:  FunctionInvocationId, ScriptHostInstanceId and HostId.

pragnagopa commented on src/WebJobs.Script/ScriptConstants.cs:
```diff
@@ -24,12 +24,19 @@ public static class ScriptConstants
         public const string TraceSourceFileWatcher = "FileWatcher";
         public const string TraceSourceSwagger = "SwaggerDocumentation";
         public const string TraceSourceHttpHandler = "HttpRequestTraceHandler";
+        public const string TraceSourceScriptHost = "Script.Host";
```
@fabiocav - Any ideas on what should be the "source" column value in our logs? Do we want to see similar to what [webjobs sdk](https://github.com/Azure/azure-webjobs-sdk/blob/663a508e8a851629c26a51e7de3af36629dfd120/src/Microsoft.Azure.WebJobs.Host/TraceSource.cs) has like "Script.Host" or should it be the class name like [this](https://github.com/Azure/azure-functions-host/blob/v1.x/src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs#L196)

mathewc commented:
What is the context on this - is there a work item/issue?

mathewc commented on src/WebJobs.Script.WebHost/Diagnostics/FunctionInstanceLogger.cs:
```diff
@@ -43,7 +45,8 @@ internal class FunctionInstanceLogger : IAsyncCollector<FunctionInstanceLogEntry
 
                 string containerName = Environment.MachineName;
-                this._writer = LogFactory.NewWriter(hostName, containerName, tableProvider, (e) => OnException(e, trace));
+                _hostId = hostInstanceId;
```
This naming is confusing. There are HostIds and HostInstanceIds. Which is this?

pragnagopa commented:
Updated the PR description to include issue<https://github.com/Azure/azure-functions-host/issues/2312>.

From: Mathew Charles <notifications@github.com>
Sent: Tuesday, February 13, 2018 12:46 PM
To: Azure/azure-functions-host <azure-functions-host@noreply.github.com>
Cc: Pragna Gopa <pgopa@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [Azure/azure-functions-host] Add columns - FunctionInvocationId, ScriptHostInstanceId and HostId (#2418)


What is the context on this - is there a work item/issue?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-functions-host%2Fpull%2F2418%23issuecomment-365398446&data=04%7C01%7Cpgopa%40microsoft.com%7C83cb58d00cb14a71badc08d57322cf8d%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636541515698043551%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwifQ%3D%3D%7C-1&sdata=ATux6BzUIZlWSnd8xgT60oSDU%2BojWIVVk9EVU4%2FmPHs%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAByZZ9n_JHcd0afSWG8q5e__nkpSRQbXks5tUfSQgaJpZM4SEUch&data=04%7C01%7Cpgopa%40microsoft.com%7C83cb58d00cb14a71badc08d57322cf8d%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636541515698043551%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwifQ%3D%3D%7C-1&sdata=Qq4oDOL4mFtPpHTbxNbgsSz7PCFi%2FYtl91euyU0pd2A%3D&reserved=0>.

pragnagopa commented:
Fixed naming and added both columns.

From: Mathew Charles <notifications@github.com>
Sent: Tuesday, February 13, 2018 12:49 PM
To: Azure/azure-functions-host <azure-functions-host@noreply.github.com>
Cc: Pragna Gopa <pgopa@microsoft.com>; Author <author@noreply.github.com>
Subject: Re: [Azure/azure-functions-host] Add columns - FunctionInvocationId, ScriptHostInstanceId and HostId (#2418)


@mathewc commented on this pull request.

________________________________

In src/WebJobs.Script.WebHost/Diagnostics/FunctionInstanceLogger.cs<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-functions-host%2Fpull%2F2418%23discussion_r167996208&data=04%7C01%7Cpgopa%40microsoft.com%7C5fde065ce72246393bb008d573233e24%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636541517625767394%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwifQ%3D%3D%7C-1&sdata=lA2d3kJp3ZYFhP5yQdbBQtDJYNl3jlSteN%2FrhVupnd8%3D&reserved=0>:

> @@ -43,7 +45,8 @@ internal class FunctionInstanceLogger : IAsyncCollector<FunctionInstanceLogEntry

                 var tableProvider = LogFactory.NewLogTableProvider(client);



                 string containerName = Environment.MachineName;

-                this._writer = LogFactory.NewWriter(hostName, containerName, tableProvider, (e) => OnException(e, trace));

+                _hostId = hostInstanceId;

This naming is confusing. There are HostIds and HostInstanceIds. Which is this?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-functions-host%2Fpull%2F2418%23pullrequestreview-96294237&data=04%7C01%7Cpgopa%40microsoft.com%7C5fde065ce72246393bb008d573233e24%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636541517625767394%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwifQ%3D%3D%7C-1&sdata=X6yMLCtSoYgewdwvVwZIbvd63PgL2x8x%2FpsfF%2BqJWVI%3D&reserved=0>, or mute the thread<https://na01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAByZZ8Evyog7oUOn5nhWCo82qhPBHknvks5tUfVJgaJpZM4SEUch&data=04%7C01%7Cpgopa%40microsoft.com%7C5fde065ce72246393bb008d573233e24%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636541517625767394%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwifQ%3D%3D%7C-1&sdata=JhQb7WTN4IqIcjMC2aUzyqgpJn0uP65dz4IMPkZFGqE%3D&reserved=0>.

mathewc commented on src/WebJobs.Script.WebHost/Diagnostics/FunctionInstanceLogger.cs:
```diff
@@ -43,7 +48,9 @@ internal class FunctionInstanceLogger : IAsyncCollector<FunctionInstanceLogEntry
 
                 string containerName = Environment.MachineName;
-                this._writer = LogFactory.NewWriter(hostName, containerName, tableProvider, (e) => OnException(e, trace));
+                _hostId = hostId;
```
Wondering why the HostID needs to be logged. We're already logging the AppName column, and in Functions we default the HostID to the AppName. How would this be used - do we actually need it?

pragnagopa commented:
Closing this PR. Synced with @mathewc. Will add new HostInstanceId for identifying ScriptHost instance. Existing InstanceId in ScriptHost.cs reflects the website_instanceid.
